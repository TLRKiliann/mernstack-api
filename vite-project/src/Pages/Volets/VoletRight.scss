/*VOLETS*/
/*Right volet*/

.volet--R {
  position: absolute;
  top: 0px;
  right: 0px;
  width: 400px;
  min-height: 100%;
  overflow: hidden;
}

@keyframes volet--rightdisplay {
  0% {
    transform: translateX(300px);
    overflow: hidden;
  }
  100% {
    transform: translateX(0px);
    overflow: hidden;
  }
}

.subvolet--R {
  position: relative;
  top: 0px;
  right: 0px;
  width: 400px;
  min-height: 120vh;
  background-color: royalblue;
  border-left: 1px solid blue;

  transform: translateX(0px);

  animation-name: volet--rightdisplay;
  animation-delay: 0s;
  animation-duration: 2s;
  animation-iteration-count: 1;
  animation-timing-function: ease;
  z-index: 5;
}

.subvolet--R button {
  margin: 10px 10px;
  width: 30px;
  height: 30px;

  border-radius: 50% 50%;
  background: dimgrey;
  color: lightgreen;
  &:hover {
    cursor: pointer;
    background: grey;
    color: whitesmoke;
  }
}

.subvolet--R p {
  margin: 20px 20px;
  padding: 5px 10px;
  text-align: center;
  background: lightblue;
  border: 2mm ridge royalblue;
  box-shadow: 0px 0px 4px whitesmoke;
  border-radius: 5px;
  color: royalblue;
}

.subvolet--R img {
  margin: -10px 0px;
  width: 60%;
  margin: auto;
  display: flex;
  background: royalblue;
  color: whitesmoke;
}

.section--voletr {
  margin: 20px 20px;
  padding: 20px 40px;
  background: lightblue;
  border: 2mm ridge royalblue;
  box-shadow: 0px 0px 4px whitesmoke;
  border-radius: 10px;
}

.li--right {
  margin-bottom: 10px;
  padding-bottom: 10px;
  border-bottom: 1px dashed blue;
  color: royalblue;
  &:hover {
    color: orange;
  }
}

.link--tagvoletr {
  text-decoration: none;
  font-weight: bold;
  color: royalblue;
}

.link--tagvoletr:hover {
  color: navy;
}


.volet--R2 {
  position: absolute;
  top: 0px;
  right: 0px;
  width: 400px;
  min-height: 100%;
  overflow: hidden;
}

.subvolet--R2 {
  position: relative;
  width: 400px;
  top: 0px;
  right: -400px;
  width: 400px;
  min-height: 120vh;
  background-color: royalblue;
  border-left: 1px solid blue;

  transform: translateX(0px);

  animation-name: volet--righthidden;
  animation-delay: 0s;
  animation-duration: 2s;
  animation-iteration-count: 1;
  animation-timing-function: ease;
  z-index: 5;
}

@keyframes volet--righthidden {
  0% {
    transform: translateX(-400px);
    overflow: hidden;
  }
  100% {
    transform: translateX(0px);
    overflow: hidden;
  }
}

.subvolet--R2 button {
  margin: 10px 10px;
  width: 30px;
  height: 30px;

  border-radius: 50% 50%;
  background: dimgrey;
  color: lightgreen;
  &:hover {
    cursor: pointer;
    background: grey;
    color: whitesmoke;
  }
}

.subvolet--R2 p {
  margin: 20px 20px;
  padding: 5px 10px;
  text-align: center;
  background: lightblue;
  border: 2mm ridge royalblue;
  box-shadow: 0px 0px 4px whitesmoke;
  border-radius: 5px;
  color: royalblue;
}

.subvolet--R2 img {
  margin: -10px 0px;
  width: 60%;
  margin: auto;
  display: flex;
  background: royalblue;
  color: whitesmoke;
}