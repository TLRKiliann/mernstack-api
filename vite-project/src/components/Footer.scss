.div--navfooter {
  position: fixed;
  right: 0px;
  bottom: 0px;
  left: 0px;
  margin: auto;
  display: flex;
  background-color: #000000;
  border-top: 1px solid slategray;
  box-shadow: 0px 0px 10px slategray;
}

.div--navfooter ul {
  width: 100vw;
  margin: auto;
  padding-left: 0px;
  display: flex;
  justify-content: center;
}

.div--navfooter li {
  width: 20vw;
  margin: auto;
  padding: 20px 0px;
  display: flex;
  justify-content: space-around;
  font-family: 'Consolase';
  font-size: 1.4rem;
  color: lightgreen;
}