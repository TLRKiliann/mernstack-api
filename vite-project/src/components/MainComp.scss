.master--maincomp {
	margin-top: 40px;
	width: 40%;
	margin: auto;
	height: auto;
}

.submaster--maincomp {
}

.bg--colormaincomp {
	width: 100%;
	margin-top: 40px;
	padding: 20px 0px;
 	border: 2mm ridge royalblue;
 	border-radius: 12px;
 	box-shadow: 0px 0px 4px whitesmoke;
	background-color: rgba(45, 85, 255, 0.4);
}

.h2--maincomp {
	text-align: center;
	font-family: 'Consolas';
	text-shadow: 2px 2px 2px blue; 
	color: lightblue;
}

.div--connection {
	width: 80%;
	padding: 5px 10px;
	margin: 10px auto;
	display: flex;
	align-items: center;
	background-color: lightblue;
 	border: 2mm ridge royalblue;
 	border-radius: 7px;
 	box-shadow: 0px 0px 4px whitesmoke;
	color: royalblue;
}