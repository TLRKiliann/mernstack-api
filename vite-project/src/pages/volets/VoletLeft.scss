/*Left volet*/
.volet--L {
  position: absolute;
  top: 0px;
  left: 0px;
  width: 400px;
  min-height: 120vh;
  overflow: hidden;
}

@keyframes volet--leftdisplay {
  0% {
    transform: translateX(-400px);
    overflow: hidden;
  }
  100% {
    transform: translateX(0px);
    overflow: hidden;
  }
}

.subvolet--L {
  position: relative;
  top: 0px;
  left: 0px;
  width: -400px;
  min-height: 120vh;
  background-color: rgba(45, 85, 255, 0.4);
  border-right: 1px solid blue;
  transform: translateX(0px);

  animation-name: volet--leftdisplay;
  animation-duration: 1s;
  animation-iteration-count: 1;
  animation-timing-function: ease;
  z-index: 11;
}

.subvolet--L button {
  margin-top: 10px;
  margin-left: 350px;
  width: 30px;
  height: 30px;

  border-radius: 50% 50%;
  background: transparent;
  color: whitesmoke;
  &:hover {
    cursor: pointer;
    background: rgba(0, 155, 255, 0.5);
    color: lightgreen;
  }
}

.subvolet--L p {
  margin: 20px 20px;
  padding: 5px 10px;
  text-align: center;
  background-color: rgba(45, 85, 255, 0.6);
  border: 2mm ridge royalblue;
  border-radius: 5px;
  font-weight: bold;
  color: whitesmoke;
}

.section--voletl {
  margin: 20px 20px;
  padding: 10px 40px;
  background-color: rgba(45, 85, 255, 0.6);
  border: 2mm ridge royalblue;
  border-radius: 10px;
}

.span {
  display: flex;
  align-items: center;
  justify-content: center;
}

.li {
  margin-bottom: 10px;
  padding-bottom: 10px;
  display: flex;
  align-items: center;
  font-weight: bold;
  border-bottom: 1px dashed whitesmoke;
}

.link--tagvoletl {
  margin: 2px 0px;
  text-decoration: none;
  font-family: 'Consolas';
  font-size: 1.2rem;
  color: whitesmoke;
  &:hover {
    color: steelblue;
  }
}

.volet--L2 {
  position: absolute;
  top: 0px;
  left: 0px;
  width: 400px;
  min-height: 120vh;
  overflow: hidden;
}

.subvolet--L2 {
  position: relative;
  top: 0px;
  left: 0px;
  width: -400px;
  min-height: 120vh;
  background-color: rgba(45, 85, 255, 0.4);
  border-right: 1px solid blue;
  transform: translateX(-400px);

  animation-name: volet--lefthidden;
  animation-duration: 1s;
  animation-iteration-count: 1;
  animation-timing-function: ease;
  z-index: 11;
}

@keyframes volet--lefthidden {
  0% {
    transform: translateX(0px);
    overflow: hidden;
  }
  100% {
    transform: translateX(-400px);
    overflow: hidden;
  }
}

.subvolet--L2 button {
  margin-top: 10px;
  margin-left: 350px;
  width: 30px;
  height: 30px;
  background: transparent;
  border-radius: 50% 50%;
  color: whitesmoke;
}

.subvolet--L2 p {
  margin: 20px 20px;
  padding: 5px 10px;
  text-align: center;
  background-color: rgba(45, 85, 255, 0.6);
  border: 2mm ridge royalblue;
  box-shadow: 0px 0px 3px whitesmoke;
  font-weight: bold;
  color: whitesmoke;
}