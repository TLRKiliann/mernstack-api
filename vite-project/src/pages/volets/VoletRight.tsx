import React from 'react'
import { Link } from 'react-router-dom'
import mother from '../../assets/background/motherboard.png'
import './VoletRight.scss'

interface VoletRightProps {
  isOpenRight: boolean
  handleVoletsRight: () => void
}

const VoletRight: React.FC = (props: VoletRightProps) => {

  return(
    <div>
      {props.isOpenRight ? (
        <div className="volet--R">
          <div className="subvolet--R">
            <button 
              onClick={props.handleVoletsRight}
              className="btn--voletr"
            >
              X
            </button>
            <p>Chat Computing</p>

            <img
              src={mother}
              width='240px'
              height='140px'
              alt='no img volet-R 1'
            />

            <section className="section--voletr">
              {props.computers?.map((computer) => (
                <li key={computer.id} className="li--right">
                  <Link className="link--tagvoletr"
                    to={`/chatcomputer/${computer.id}`}
                  >
                    {computer.title}
                  </Link>
                </li>
                ))
              }
            </section>

          </div>
        </div>
        ) : (
        <div className="volet--R2">
          <div className="subvolet--R2">

            <button 
              onClick={props.handleVoletsRight}
              className="btn--voletr"
            >
              X
            </button>
            <p>Chat Rooms</p>

            <img
              src={mother}
              width='240px'
              height='140px'
              alt='no img volet-R 2'
            />

            <section className="section--voletr">
              {props.computers?.map((computer) => (
                <li key={computer.id} className="li--right">
                  <Link className="link--tagvoletr"
                    to={`/chatcomputer/${computer.id}`}
                  >
                    {computer.title}
                  </Link>
                </li>
                ))
              }
            </section>

          </div>
        </div>
        )
      }
    </div>
  )
}

export default VoletRight