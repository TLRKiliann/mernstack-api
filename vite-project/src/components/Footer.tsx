import { Component } from 'react'
import './styleComponents/Footer.scss'


export default class Footer extends Component {
  render() {
    return(
      <div className="div--navfooter">
        <nav className="navbar--nav">
          <ul className="navbar--ul">
            <li className="navbar--li">
              Contact
            </li>
            <li className="navbar--li">
              About US
            </li>
            <li className="navbar--li">
              Rules
            </li>
          </ul>
        </nav>
      </div>
    )
  }
}