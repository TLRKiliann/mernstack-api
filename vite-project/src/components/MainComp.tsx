import React from 'react'
import { userType } from '../models/userType';
import './styleComponents/MainComp.scss'

interface MainCompProps {
  users: userType[]
}

const MainComp: React.FC = (props: MainCompProps) => {
  return(
    <div className="master--maincomp">
          
      <div className="bg--colormaincomp">

        <h2 className='h2--maincomp'>
          All Members :&nbsp;
          <span style={{color: "orange"}}>
            {props.users?.length}
          </span>
        </h2>
        
        <h3 className="h3--maincomp">
          Users online (connected) :&nbsp;
          <span
            style={{color:'whitesmoke'}}
          >
            {props.users?.reduce((prev, curr) =>
            prev + curr.isConnected, 0)}
          </span>
        </h3>

        <div>
          {props.users?.slice(0, `${props.users?.length}`).map(user => (
            <div key={user.id}>
            {user.isConnected ? (
              <div className="div--connection">
                <img
                  src={user.img}
                  width="40px"
                  height="40px"
                  className="img--connector"
                  alt={user.img}
                />
                {user.firstName} {user.lastName} :&nbsp;
                  <div 
                    style={{
                      fontFamily: 'Junction',
                      fontSize: '1.2rem',
                      color: 'lightgreen'
                    }}>
                    Connected
                  </div>
              </div>
              ) : null
            }
            </div>
            ))
          }
        </div>
      </div>
    </div>
  )
}

export default MainComp