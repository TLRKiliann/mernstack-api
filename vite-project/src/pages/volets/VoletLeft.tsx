import React, { useState } from 'react'
import { Link } from 'react-router-dom'
import { MdConnectWithoutContact } from 'react-icons/md'
import { TbPlugConnected } from 'react-icons/tb'
import { BsFillChatDotsFill } from 'react-icons/bs'
import './VoletLeft.scss'


interface VoletLeftProps {
  isOpenL: boolean
  handleVoletsLeft: () => void
}

const VoletLeft: React.FC = (props: VoletLeftProps) => {

  const [searchUser, setSearchUser] = useState<Array<string>>([])
  const [userFound, setUserFound] = useState<Array<string>>([])
  //console.log(searchUser)

  const handleSearchInput = (e: React.ChangeEvent<HTMLIputElement>): void => {
    setSearchUser(e.target.value)
  }

  const handleSearchUser = (e: React.MouseEvent<HTMLButtonElement>) => {
    const findByFirstName = props.users?.map(user => user).filter(user => {
      return user.firstName === searchUser
        ? `${user.firstname} ${user.lastName}`
        : null
    })
    e.preventDefault();
    if (searchUser === "") {
      setSearchUser("");
      alert("User undefined or not existing...")
    } else {
      setUserFound(findByFirstName)
      setSearchUser("")
    }
  };

  return(
    <div>
      {props.isOpenL ? (
        <div className="volet--L">
          <div className="subvolet--L">
            <button 
              className="btn--volet"
              onClick={props.handleVoletsLeft}>X</button>
            <p>Private Chat</p>
              
            <span className="span">
              <BsFillChatDotsFill size={64}/>
            </span>

            <div className="form--search">
              <input
                type="text"
                value={searchUser}
                onChange={(e) => handleSearchInput(e)}
                placeholder="Search by firstname"
                className="search--input"
              />
              <button
                onClick={handleSearchUser}
                className="btn--search"
              >
                Enter
              </button>
            </div>

            
            <div className="result--search">
            {userFound?.map(userFirst => (
              <p key={userFirst.id}
                className="result--searchp">{userFirst.firstName} {userFirst.lastName} 
                <span 
                  style={userFirst.isConnected ? {color: 'lightgreen'} 
                    : {color: 'orange'}}
                >
                  &nbsp;{userFirst.isConnected ? "Connected" : "No Connected"}
                </span>
              </p>
            ))}
            </div>


            <section className="section--voletl">
              {props.users?.map(user => (
                <li key={user.id} className="li"
                  style={user.isConnected 
                  ? {color: 'lightgreen'}
                  : {color: 'orange'}}
                >
                  {user.isConnected ? (
                    <MdConnectWithoutContact
                      size={24}
                      style={{
                        marginRight: '10px',
                      }}
                    />
                    ) : (
                    <TbPlugConnected
                      size={24}
                      style={{marginRight: '10px'}}
                    />
                    )
                  }
                  <Link className="link--tagvoletl"
                    to={`/computerroom/privatemessage/${user.firstName}`}
                  >
                    {user.firstName} {user.lastName}
                  </Link>
                </li>
                ))
              }
            </section>
          
          </div>
        </div>
        
        ) : (

        <div className="volet--L2">
          <div className="subvolet--L2">
            <button
              className="btn--volet"
              onClick={props.handleVoletsLeft}>X</button>
            <p>Users State Connection</p>

            <span className="span">
              <BsFillChatDotsFill size={64}/>
            </span>

            <div className="form--search">
              <input
                type="text"
                value={searchUser}
                onChange={(e) => handleSearchInput(e)}
                placeholder="Search by firstname"
                className="search--input"
              />
              <button
                onClick={handleSearchUser}
                className="btn--search"
              >
                Enter
              </button>
            </div>

            <section className="section--voletl">
              {props.users?.map(user => (
                <li key={user.id} className="li">
                  {user.isConnected ? (
                    <MdConnectWithoutContact
                      size={24}
                      style={{marginRight: '10px'}}
                    />
                    ) : (
                    <TbPlugConnected
                      size={24}
                      style={{marginRight: '10px'}}
                    />
                    )
                  }
                  <Link className="link--tagvoletl"
                    to={`/computerroom/privatemessage/${user.firstName}`}
                  >
                    {user.firstName} {user.lastName}
                  </Link>
                </li>
                ))
              }
            </section>

          </div>
        </div>
        )
      }
    </div>
  )
}

export default VoletLeft