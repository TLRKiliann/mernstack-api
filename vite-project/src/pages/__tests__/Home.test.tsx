import React from "react"
import {assert, afterEach, describe, expect, test, it, vi} from 'vitest'
//import {fireEvent, render, screen} from '@testing-library/react'
import { create } from 'react-test-renderer'
import VoletLeft from '../volets/VoletLeft'
import VoletRight from '../volets/VoletRight'
import MainTitle from '../../components/MainTitle'
import MainComp from '../../components/MainComp'
import Footer from '../../components/Footer'
//import "@testing-library/jest-dom/extend-expect"
//import "@testing-library/jest-dom";
//import { act } from 'react-dom/test-utils';
//assertType, expectTypeOf

test('MatchSnapShot test VoletLeft', () => {
  const treeVL = create(<VoletLeft />)
  expect(treeVL.toJSON()).toMatchSnapshot();
})

test('MatchSnapShot test VoletRight', () => {
  const treeVR = create(<VoletRight />)
  expect(treeVR.toJSON()).toMatchSnapshot();
})

test('MatchSnapShot test MainTitle', () => {
  const treeMainTitle = create(<MainTitle />)
  expect(treeMainTitle.toJSON()).toMatchSnapshot()
})

test('MatchSnapShot test MainComp', () => {
  const treeMainComp = create(<MainComp />)
  expect(treeMainComp.toJSON()).toMatchSnapshot()
})

test('MatchSnapShot test Footer 1', () => {
  const tree2 = create(<Footer />)
  expect(tree2.toJSON()).toMatchSnapshot();
})

describe('suite name', () => {
  it('foo', () => {
    assert.equal(Math.sqrt(4), 2)
  })

  it('bar', () => {
    expect(1 + 1).eq(2)
  })

  it('snapshot', () => {
    expect({ foo: 'bar' }).toMatchSnapshot()
  })
})

const getLatest = (index = messages.items.length - 1) => messages.items[index]

const messages = {
  items: [
    { message: 'Simple test message', from: 'Testman' },
    // ...
  ],
  getLatest, // can also be a `getter or setter if supported`
}

describe('reading messages', () => {
  afterEach(() => {
    vi.restoreAllMocks()
  })

  it('should get the latest message with a spy', () => {
    const spy = vi.spyOn(messages, 'getLatest')
    expect(spy.getMockName()).toEqual('getLatest')

    expect(messages.getLatest()).toEqual(
      messages.items[messages.items.length - 1],
    )

    expect(spy).toHaveBeenCalledTimes(1)

    spy.mockImplementationOnce(() => 'access-restricted')
    expect(messages.getLatest()).toEqual('access-restricted')

    expect(spy).toHaveBeenCalledTimes(2)
  })

  it('should get with a mock', () => {
    const mock = vi.fn().mockImplementation(getLatest)

    expect(mock()).toEqual(messages.items[messages.items.length - 1])
    expect(mock).toHaveBeenCalledTimes(1)

    mock.mockImplementationOnce(() => 'access-restricted')
    expect(mock()).toEqual('access-restricted')

    expect(mock).toHaveBeenCalledTimes(2)

    expect(mock()).toEqual(messages.items[messages.items.length - 1])
    expect(mock).toHaveBeenCalledTimes(3)
  })
})