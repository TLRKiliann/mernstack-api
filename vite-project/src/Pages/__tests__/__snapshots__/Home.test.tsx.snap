// Vitest Snapshot v1

exports[`MatchSnapShot test Footer 1 1`] = `
<div>
  <div
    className="wallIntro"
  >
    <h1>
      Wellcome To ChatRoom !
    </h1>
  </div>
  <div
    className="btn--voletsLeft"
  >
    <button
      onClick={[Function]}
    >
      Click me(L)
    </button>
  </div>
  <div
    className="btn--voletsRight"
  >
    <button
      onClick={[Function]}
    >
      Click me(R)
    </button>
  </div>
  <div
    className="volet--R2"
  >
    <div
      className="subvolet--R2"
    >
      <button
        onClick={[Function]}
      >
        X
      </button>
      <p>
        Volet R 2
      </p>
    </div>
  </div>
  <div
    className="volet--L2"
  >
    <div
      className="subvolet--L2"
    >
      <p>
        Volet L 2
      </p>
      <button
        onClick={[Function]}
      >
        Close
      </button>
    </div>
  </div>
  <div
    style={
      {
        "background": "#161616",
        "border": "1px solid lightgreen",
        "borderRadius": "10px",
        "boxShadow": "0px 0px 10px lightgreen",
        "color": "whitesmoke",
        "fontFamily": "Consolas, serif",
        "height": "auto",
        "margin": "auto",
        "marginTop": "100px",
        "textAlign": "center",
        "width": "600px",
      }
    }
  >
    <h1>
      Chat-Room STACK
    </h1>
    <h2>
      Text Header
    </h2>
    <h4>
      MY SECOND TEXT HEADER
    </h4>
  </div>
  <div
    style={
      {
        "background": "#161616",
        "color": "lightgreen",
        "fontSize": "1.2rem",
        "margin": "auto",
        "marginTop": "40px",
        "padding": "20px 40px",
        "width": "600px",
      }
    }
  >
    <h2
      style={
        {
          "textAlign": "center",
        }
      }
    >
      All Members !
    </h2>
    <div>
      <p>
        Pierre
         
        Dujardin
         - 
        44
         ans - 
        male
      </p>
      <p>
        Celestine
         
        Chakra
         - 
        37
         ans - 
        female
      </p>
      <p>
        Gérard
         
        Golan
         - 
        28
         ans - 
        male
      </p>
      <p>
        Juju
         
        Ptitcoeur
         - 
        42
         ans - 
        female
      </p>
      <p>
        Nathi
         
        Naelle
         - 
        34
         ans - 
        female
      </p>
    </div>
    <span />
  </div>
  <div
    className="div--review"
  >
    <h1>
      Reports
    </h1>
    <p>
      Pierre
       
      43
    </p>
    <p>
      Celestine
       
      3
    </p>
    <p>
      Gérard
       
      14
    </p>
    <p>
      Juju
       
      21
    </p>
    <p>
      Nathi
       
      99
    </p>
  </div>
  <div
    className="div--navfooter"
  >
    <nav>
      <ul>
        <li>
          Contact
        </li>
        <li>
          About US
        </li>
        <li>
          Rules
        </li>
      </ul>
    </nav>
  </div>
</div>
`;

exports[`MatchSnapShot test Header 1 1`] = `
<div>
  <div
    className="wallIntro"
  >
    <h1>
      Wellcome To ChatRoom !
    </h1>
  </div>
  <div
    className="btn--voletsLeft"
  >
    <button
      onClick={[Function]}
    >
      Click me(L)
    </button>
  </div>
  <div
    className="btn--voletsRight"
  >
    <button
      onClick={[Function]}
    >
      Click me(R)
    </button>
  </div>
  <div
    className="volet--R2"
  >
    <div
      className="subvolet--R2"
    >
      <button
        onClick={[Function]}
      >
        X
      </button>
      <p>
        Volet R 2
      </p>
    </div>
  </div>
  <div
    className="volet--L2"
  >
    <div
      className="subvolet--L2"
    >
      <p>
        Volet L 2
      </p>
      <button
        onClick={[Function]}
      >
        Close
      </button>
    </div>
  </div>
  <div
    style={
      {
        "background": "#161616",
        "border": "1px solid lightgreen",
        "borderRadius": "10px",
        "boxShadow": "0px 0px 10px lightgreen",
        "color": "whitesmoke",
        "fontFamily": "Consolas, serif",
        "height": "auto",
        "margin": "auto",
        "marginTop": "100px",
        "textAlign": "center",
        "width": "600px",
      }
    }
  >
    <h1>
      Chat-Room STACK
    </h1>
    <h2>
      Text Header
    </h2>
    <h4>
      MY SECOND TEXT HEADER
    </h4>
  </div>
  <div
    style={
      {
        "background": "#161616",
        "color": "lightgreen",
        "fontSize": "1.2rem",
        "margin": "auto",
        "marginTop": "40px",
        "padding": "20px 40px",
        "width": "600px",
      }
    }
  >
    <h2
      style={
        {
          "textAlign": "center",
        }
      }
    >
      All Members !
    </h2>
    <div>
      <p>
        Pierre
         
        Dujardin
         - 
        44
         ans - 
        male
      </p>
      <p>
        Celestine
         
        Chakra
         - 
        37
         ans - 
        female
      </p>
      <p>
        Gérard
         
        Golan
         - 
        28
         ans - 
        male
      </p>
      <p>
        Juju
         
        Ptitcoeur
         - 
        42
         ans - 
        female
      </p>
      <p>
        Nathi
         
        Naelle
         - 
        34
         ans - 
        female
      </p>
    </div>
    <span />
  </div>
  <div
    className="div--review"
  >
    <h1>
      Reports
    </h1>
    <p>
      Pierre
       
      43
    </p>
    <p>
      Celestine
       
      3
    </p>
    <p>
      Gérard
       
      14
    </p>
    <p>
      Juju
       
      21
    </p>
    <p>
      Nathi
       
      99
    </p>
  </div>
  <div
    className="div--navfooter"
  >
    <nav>
      <ul>
        <li>
          Contact
        </li>
        <li>
          About US
        </li>
        <li>
          Rules
        </li>
      </ul>
    </nav>
  </div>
</div>
`;

exports[`MatchSnapShot test MainComp 1 1`] = `
<div>
  <div
    className="wallIntro"
  >
    <h1>
      Wellcome To ChatRoom !
    </h1>
  </div>
  <div
    className="btn--voletsLeft"
  >
    <button
      onClick={[Function]}
    >
      Click me(L)
    </button>
  </div>
  <div
    className="btn--voletsRight"
  >
    <button
      onClick={[Function]}
    >
      Click me(R)
    </button>
  </div>
  <div
    className="volet--R2"
  >
    <div
      className="subvolet--R2"
    >
      <button
        onClick={[Function]}
      >
        X
      </button>
      <p>
        Volet R 2
      </p>
    </div>
  </div>
  <div
    className="volet--L2"
  >
    <div
      className="subvolet--L2"
    >
      <p>
        Volet L 2
      </p>
      <button
        onClick={[Function]}
      >
        Close
      </button>
    </div>
  </div>
  <div
    style={
      {
        "background": "#161616",
        "border": "1px solid lightgreen",
        "borderRadius": "10px",
        "boxShadow": "0px 0px 10px lightgreen",
        "color": "whitesmoke",
        "fontFamily": "Consolas, serif",
        "height": "auto",
        "margin": "auto",
        "marginTop": "100px",
        "textAlign": "center",
        "width": "600px",
      }
    }
  >
    <h1>
      Chat-Room STACK
    </h1>
    <h2>
      Text Header
    </h2>
    <h4>
      MY SECOND TEXT HEADER
    </h4>
  </div>
  <div
    style={
      {
        "background": "#161616",
        "color": "lightgreen",
        "fontSize": "1.2rem",
        "margin": "auto",
        "marginTop": "40px",
        "padding": "20px 40px",
        "width": "600px",
      }
    }
  >
    <h2
      style={
        {
          "textAlign": "center",
        }
      }
    >
      All Members !
    </h2>
    <div>
      <p>
        Pierre
         
        Dujardin
         - 
        44
         ans - 
        male
      </p>
      <p>
        Celestine
         
        Chakra
         - 
        37
         ans - 
        female
      </p>
      <p>
        Gérard
         
        Golan
         - 
        28
         ans - 
        male
      </p>
      <p>
        Juju
         
        Ptitcoeur
         - 
        42
         ans - 
        female
      </p>
      <p>
        Nathi
         
        Naelle
         - 
        34
         ans - 
        female
      </p>
    </div>
    <span />
  </div>
  <div
    className="div--review"
  >
    <h1>
      Reports
    </h1>
    <p>
      Pierre
       
      43
    </p>
    <p>
      Celestine
       
      3
    </p>
    <p>
      Gérard
       
      14
    </p>
    <p>
      Juju
       
      21
    </p>
    <p>
      Nathi
       
      99
    </p>
  </div>
  <div
    className="div--navfooter"
  >
    <nav>
      <ul>
        <li>
          Contact
        </li>
        <li>
          About US
        </li>
        <li>
          Rules
        </li>
      </ul>
    </nav>
  </div>
</div>
`;

exports[`MatchSnapShot test MainTitle 1 1`] = `
<div>
  <div
    className="wallIntro"
  >
    <h1>
      Wellcome To ChatRoom !
    </h1>
  </div>
  <div
    className="btn--voletsLeft"
  >
    <button
      onClick={[Function]}
    >
      Click me(L)
    </button>
  </div>
  <div
    className="btn--voletsRight"
  >
    <button
      onClick={[Function]}
    >
      Click me(R)
    </button>
  </div>
  <div
    className="volet--R2"
  >
    <div
      className="subvolet--R2"
    >
      <button
        onClick={[Function]}
      >
        X
      </button>
      <p>
        Volet R 2
      </p>
    </div>
  </div>
  <div
    className="volet--L2"
  >
    <div
      className="subvolet--L2"
    >
      <p>
        Volet L 2
      </p>
      <button
        onClick={[Function]}
      >
        Close
      </button>
    </div>
  </div>
  <div
    style={
      {
        "background": "#161616",
        "border": "1px solid lightgreen",
        "borderRadius": "10px",
        "boxShadow": "0px 0px 10px lightgreen",
        "color": "whitesmoke",
        "fontFamily": "Consolas, serif",
        "height": "auto",
        "margin": "auto",
        "marginTop": "100px",
        "textAlign": "center",
        "width": "600px",
      }
    }
  >
    <h1>
      Chat-Room STACK
    </h1>
    <h2>
      Text Header
    </h2>
    <h4>
      MY SECOND TEXT HEADER
    </h4>
  </div>
  <div
    style={
      {
        "background": "#161616",
        "color": "lightgreen",
        "fontSize": "1.2rem",
        "margin": "auto",
        "marginTop": "40px",
        "padding": "20px 40px",
        "width": "600px",
      }
    }
  >
    <h2
      style={
        {
          "textAlign": "center",
        }
      }
    >
      All Members !
    </h2>
    <div>
      <p>
        Pierre
         
        Dujardin
         - 
        44
         ans - 
        male
      </p>
      <p>
        Celestine
         
        Chakra
         - 
        37
         ans - 
        female
      </p>
      <p>
        Gérard
         
        Golan
         - 
        28
         ans - 
        male
      </p>
      <p>
        Juju
         
        Ptitcoeur
         - 
        42
         ans - 
        female
      </p>
      <p>
        Nathi
         
        Naelle
         - 
        34
         ans - 
        female
      </p>
    </div>
    <span />
  </div>
  <div
    className="div--review"
  >
    <h1>
      Reports
    </h1>
    <p>
      Pierre
       
      43
    </p>
    <p>
      Celestine
       
      3
    </p>
    <p>
      Gérard
       
      14
    </p>
    <p>
      Juju
       
      21
    </p>
    <p>
      Nathi
       
      99
    </p>
  </div>
  <div
    className="div--navfooter"
  >
    <nav>
      <ul>
        <li>
          Contact
        </li>
        <li>
          About US
        </li>
        <li>
          Rules
        </li>
      </ul>
    </nav>
  </div>
</div>
`;

exports[`MatchSnapShot test Review 1 1`] = `
<div>
  <div
    className="wallIntro"
  >
    <h1>
      Wellcome To ChatRoom !
    </h1>
  </div>
  <div
    className="btn--voletsLeft"
  >
    <button
      onClick={[Function]}
    >
      Click me(L)
    </button>
  </div>
  <div
    className="btn--voletsRight"
  >
    <button
      onClick={[Function]}
    >
      Click me(R)
    </button>
  </div>
  <div
    className="volet--R2"
  >
    <div
      className="subvolet--R2"
    >
      <button
        onClick={[Function]}
      >
        X
      </button>
      <p>
        Volet R 2
      </p>
    </div>
  </div>
  <div
    className="volet--L2"
  >
    <div
      className="subvolet--L2"
    >
      <p>
        Volet L 2
      </p>
      <button
        onClick={[Function]}
      >
        Close
      </button>
    </div>
  </div>
  <div
    style={
      {
        "background": "#161616",
        "border": "1px solid lightgreen",
        "borderRadius": "10px",
        "boxShadow": "0px 0px 10px lightgreen",
        "color": "whitesmoke",
        "fontFamily": "Consolas, serif",
        "height": "auto",
        "margin": "auto",
        "marginTop": "100px",
        "textAlign": "center",
        "width": "600px",
      }
    }
  >
    <h1>
      Chat-Room STACK
    </h1>
    <h2>
      Text Header
    </h2>
    <h4>
      MY SECOND TEXT HEADER
    </h4>
  </div>
  <div
    style={
      {
        "background": "#161616",
        "color": "lightgreen",
        "fontSize": "1.2rem",
        "margin": "auto",
        "marginTop": "40px",
        "padding": "20px 40px",
        "width": "600px",
      }
    }
  >
    <h2
      style={
        {
          "textAlign": "center",
        }
      }
    >
      All Members !
    </h2>
    <div>
      <p>
        Pierre
         
        Dujardin
         - 
        44
         ans - 
        male
      </p>
      <p>
        Celestine
         
        Chakra
         - 
        37
         ans - 
        female
      </p>
      <p>
        Gérard
         
        Golan
         - 
        28
         ans - 
        male
      </p>
      <p>
        Juju
         
        Ptitcoeur
         - 
        42
         ans - 
        female
      </p>
      <p>
        Nathi
         
        Naelle
         - 
        34
         ans - 
        female
      </p>
    </div>
    <span />
  </div>
  <div
    className="div--review"
  >
    <h1>
      Reports
    </h1>
    <p>
      Pierre
       
      43
    </p>
    <p>
      Celestine
       
      3
    </p>
    <p>
      Gérard
       
      14
    </p>
    <p>
      Juju
       
      21
    </p>
    <p>
      Nathi
       
      99
    </p>
  </div>
  <div
    className="div--navfooter"
  >
    <nav>
      <ul>
        <li>
          Contact
        </li>
        <li>
          About US
        </li>
        <li>
          Rules
        </li>
      </ul>
    </nav>
  </div>
</div>
`;

exports[`suite name > snapshot 1`] = `
{
  "foo": "bar",
}
`;
